### DISCUSSION TITLE
Any plans to support partial hydration?

### DISCUSSION THREAD

**Discussion by kawpii** [gatsbyjs/gatsby#35577] [Ideas / Feature Requests] [answered]
Astro have partial hydration which make it attractive. Any plans to support partial hydration in Gatsby?

**Comment by kawpii** [comment]
@leerob Any plan to have partial hydration like Astro? Not happy with Gatsby shipping JS unnecessarily, it goes against one of Gatsby's core goal, performance.

**Comment by tujoworker** [comment, marked as answer]
@kaushalyap [Here](https://github.com/gatsbyjs/gatsby/pull/24903#issuecomment-1192315575) I have seen that @LekoArts (the most amazing person on earth) has mentioned that there will come an RFC on this topic.

**Reply by kawpii** [reply]
@tujoworker Actually decided to leave Gatsby for Astro. Due to following reasons:

- Slow builds, wish they moved to Vite or SWC.
- Dependency bloat : Recent on Gatsby sub user pointed out that [Gatsby use lots of dependencies](https://www.npmjs.com/package/gatsby) [compared to Next.](https://www.npmjs.com/package/next)

**Reply by tujoworker** [reply]
> Dependency bloat

If you would look closely, you would have spot that Next uses 10 more dependencies. It does not matter if they are declared as dev or not. They get installed anyway.

**Reply by kawpii** [reply]
> It does not matter if they are declared as dev or not. They get installed anyway.

But they do not get into prod bundle.

**Reply by tujoworker** [reply]
Sorry to say, but that’s not how it works. It does not matter where they are declared. What gets shipped to the end-user happens in many other places, but not in package.json

**Reply by leerob** [reply]
> If you would look closely, you would have spot that Next uses 10 more dependencies. It does not matter if they are declared as dev or not. They get installed anyway.

👋 Lee from the Next.js team. To clarify, yes we do inline dependencies inside of Next.js so there is faster package installation and improved security. However, we also try to strip out extra dependencies wherever possible. 

https://packagephobia.com/result?p=next

**Reply by wardpeet** [reply]
@leerob has a point. Inlining deps does make installation faster and at some point, we will start implementing something as well for some of our deps but it doesn't really matter for your end user. We only bundle what's necessary.

 Improved security - I have to disagree with this because you can still vendor a package that has a vulnerability.

**Reply by leerob** [reply]
> Improved security - I have to disagree with this because you can still vendor a package that has a vulnerability.

I'm not saying it prevents that 😄 Of course, you can still vendor a package that has a vulnerability. But it reduces one attack vector where you `npm i` and the new, malicious package has been published.
